tests/reconfig/setup.md

```diff
@@ -0,0 +1,96 @@
+# Reconfig tests
+
+## Setup
+
+1. Create cloud cluster
+2. Deploy CRDs:
+
+   ```bash
+   kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v0.8.1/standard-install.yaml
+   ```
+
+3. Deploy NGF from edge using Helm install (NOTE: For Test 1, deploy AFTER resources):
+
+   ```bash
+   helm install my-release oci://ghcr.io/nginxinc/charts/nginx-gateway-fabric  --version 0.0.0-edge \
+      --create-namespace --wait -n nginx-gateway
+   ```
+
+4. Optional: Add pod scrape if running in GKE (see [GKE Pod scrape config](#gke-pod-scrape-config)).
+5. Run tests:
+   1. There are 3 versions of the reconfiguration tests that need to be ran, with a low and high number of resources.
+      Therefore, a full test suite includes 6 test runs.
+   2. There are scripts to generate the required resources and config changes.
+   3. Run each test using the provided script (`scripts/create-resources-gw-last.sh` or
+      `scripts/create-resources-routes-last.sh` depending on the test).
+   4. The scripts accept a number parameter to indicate how many resources should be created. Currently, we are running
+      with 30 or 150. The scripts will create a single Gateway, Secret and ReferenceGrant resources, `x+1` number of
+      namespaces, `2x` number of backend apps and services, and `3x` number of HTTPRoutes.
+      - Note: Clean up after each test run for isolated results. There's a script provided for removing all the test
+        fixtures `scripts/delete-multiple.sh` which takes a number (needs to be the same number as what was used in the
+        create script.)
+6. After each individual test run, grab logs of both NGF containers and grab metrics.
+   Note: You can expose metrics by running the below snippet and then navigating to `127.0.0.1:9113/metrics`:
+
+   ```bash
+   GW_POD=$(k get pods -n nginx-gateway | sed -n '2s/^\([^[:space:]]*\).*$/\1/p')
+   kubectl port-forward $GW_POD -n nginx-gateway 9113:9113 &
+   ```
+
+7. Measure Time To Ready by as described in each test, get the reload count, and get the average NGINX reload duration.
```

Comment:
I think we're missing an important data point about how long it take for statuses to get updated.

There are two main outputs of the system:
- updated NGINX configuration with NGINX being reloaded (we check for that)
- updated statuses (we don't check for that)

Statuses tell users (CI/CD pipelines or humans) that resources have been successfully processed and applied, which is a trigger for them to take actions - for example, roll back in case of problems or proceed further in their steps in case of success.

A few ideas here:
- adding debug logging that could print something like `"Start handling events"` and `"End handling events", "totalTime", "30s"` here https://github.com/nginxinc/nginx-gateway-fabric/blob/0c043de95e66474c9418025c9cb29e4d390a9038/internal/mode/static/handler.go#L71-L117
- introducing histogram metrics for HandleEventBatch function, how long it takes.

I'd like to be mindful about the scope of this test. But at the same time, I think we're missing out on catching some possible issues. For example, if reload time is < 353 ms in all tests, but status updates might take 10 mins, we have a problem. Note that during status updates, our control plane cannot handle any other changes to cluster resources, so this means control plane cannot react to changes in the cluster. Considering that only the leader report statuses, this also means configuration skew among replicas.

Reply:
We talked offline.
We will not handle this in this PR. but tackle it separately.